File: hardware-mall-backend/src/main/java/com/example/mystore/entity/db/Logistics.java
package com.example.mystore.entity.db;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName(value = "logistics", autoResultMap = true)
public class Logistics {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String name;
    
    private String code;
    
    private String description;
    
    private String contact;
    
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> phones;

    private String city;
    
    private String address;

    private Integer sortOrder;
    
    private Integer status;
    
    private LocalDateTime createTime;
    
    private LocalDateTime updateTime;
    
    private Long deleteTime;
}

File: hardware-mall-admin/src/api/admin/logistics.ts
import request from '@/utils/request'

export interface Logistics {
  id?: number
  name: string
  code: string
  description?: string
  contact?: string
  phone?: string
  phones?: string[]
  city?: string
  address?: string
  sortOrder?: number
  status?: number
  createTime?: string
  updateTime?: string
  deleteTime?: number
}

export interface LogisticsQuery {
  page?: number
  limit?: number
  name?: string
  city?: string
  status?: number
}

export const getLogisticsList = (params?: LogisticsQuery) => {
  return request.get<any>('/admin/logistics/list', { params })
}

export const getLogisticsById = (id: number) => {
  return request.get<Logistics>(`/admin/logistics/${id}`)
}

export const createLogistics = (data: Logistics) => {
  return request.post<Logistics>('/admin/logistics', data)
}

export const updateLogistics = (id: number, data: Logistics) => {
  return request.put<Logistics>(`/admin/logistics/${id}`, data)
}

export const deleteLogistics = (id: number) => {
  return request.delete(`/admin/logistics/${id}`)
}

export const updateLogisticsStatus = (id: number, status: number) => {
  return request.put(`/admin/logistics/${id}/status`, { status })
}
